### DISCUSSION TITLE
Question: AVB Frame size Limit

### DISCUSSION THREAD

**Discussion by KimHyeri0311** [CoRE-RG/CoRE4INET#33] [Q&A] [answered]
Hello, I have a question regarding to AVB 802.1Qav
I want to send a Stream A (Period 16.66 ms, Size 43380Bytes, Class A) and this stream should be split into multiple frames.
The maximum Length of Ethernet Frame payload should be 1500 Bytes, then I can manually split this stream into 64 x (678 Bytes payload frame) 
But when I generate the network by using ANDL example, traffic configuration is automatically set to 326 Bytes with intervalFrames parameter 1

Is the configuration(manual) below correct? 

**This code is for Configuration of Node.ini**
**.Cam3_App.numApps = 1

**.Cam3_App.app[0].typename = "AVBCustomIntervalTrafficSourceApp"
**.Cam3_App.app[0].streamID = 25
**.Cam3_App.app[0].srClass = "A"
**.Cam3_App.app[0].intervalFrames = 64
**.Cam3_App.app[0].interval = uniform(0.01666s, 0.01666s)
**.Cam3_App.app[0].payload = 678Byte

**Comment by Musteblume** [comment, marked as answer]
Hi,

the ANDL generates a default AVBTrafficSourceApp. The default interval in AVB for SRclass A is 125us (B is 250us).
16,66ms / 125us = 133,28
133,28 * 326Byte = 43.449,28Byte
You are using the AVBCustomIntervalTrafficSourceApp where you can specify your custom interval of 16,66ms (uniform is just needed if you have a time span).
So, both configurations transport the same amount of traffic.

There is one difference that must be considered.
Larger frames at same bandwidth result in different behavior of the CBS credit.
After sending a frame with 678Byte the credit will not increase back to 0 within the SRclasss A class measurement interval (also 125us).
This results in an increased worst case queueing delay for concurrent SRclass A traffic.

Hope this answers your question.

Best regards
Philipp

**Comment by KimHyeri0311** [comment]
Hi,
Thank you for your quick answer and the explanation. It helped me a lot!

Best Regards,
Haeri Kim
